genai/functions.go

```diff
@@ -0,0 +1,186 @@
+package genai
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	pb "cloud.google.com/go/ai/generativelanguage/apiv1beta/generativelanguagepb"
+	"github.com/google/generative-ai-go/internal/support"
+)
+
+// A Tool is a piece of code that enables the system to interact with
```

Comment:
It's Tool in file `functions.go` but talking about functions - do we just use the two interchangeably? Aligning with how the Python SDK names things?

Reply:
Both OpenAI and Gemini group functions into "tools." The Python SDK follows that, and we do too.

Reply:
Right, so maybe we should structure it like that too - tool is the high level concept, and functions are one of the tools available. So for example the `Tool` type would be in `tools.go`, not `functions.go`?

Also, is containment the right paradigm here? Is a function a kind of tool, or do tools contain functions and potentially other stuff?

Reply:
Renamed the file.
In the protos, a Tool contains FunctionDeclarations. Python keeps that structure.